ERCS/erc-7548.md

```diff
@@ -0,0 +1,180 @@
+// SPDX-License-Identifier: CC0-1.0
+pragma solidity ^0.8.10;
+
+import "./IERC721X.sol";
+
+interface ILicense {
+    /// @dev Verify the permission when minting a combo
+    /// @param user The minter
+    /// @param combo The new NFT to be minted by remixing multiple existing NFTs
+    /// @return components The multiple existing NFTs used to mint the new combo
+    function verify(
+        address user,
+        IERC721X.Token calldata combo,
+        IERC721X.Component[] calldata components
+    ) external returns (bool);
+}
+```
+
+### Network Module
+
+This module follows the singleton pattern and is used to track all relationships between the original NFTs and their NFT derivative works.
+
+```solidity
+// SPDX-License-Identifier: CC0-1.0
+pragma solidity ^0.8.10;
+
+import "./IERC721X.sol";
+
+interface INFTNetIndexer {
+    /// @dev Verify if the `child` was created by remixing the `parent` with other NFTs.
+    /// @param parent Any NFT
+    /// @param child Any NFT
+    function isParent(
+        IERC721X.Token calldata parent,
+        IERC721X.Token calldata child
+    ) external view returns (bool);
+
+    /// @dev Verify if `a` and `b` have common `parent`s
+    /// @param a Any NFT
+    /// @param b Any NFT
+    function isSibling(
+        IERC721X.Token calldata a,
+        IERC721X.Token calldata b
+    ) external view returns (bool, IERC721X.Token[] memory commonParents);
+
+    /// @dev Return all parents of a `token`
+    /// @param token Any NFT
+    /// @return parents All NFTs used to mint the `token`
+    function getParents(
+        IERC721X.Token calldata token
+    ) external view returns (IERC721X.Token[] memory parents);
+}
+```
+
+## Rationale
+
+The Open IP Protocol is built on the "1 premise, 2 extensions, 1 constant" principle.  
+
+The “1 premise” means that for any IP in the Open IP ecosystem, an NFT stands for that IP. So, if you have the NFT, you own the IP. That’s why the Open IP Protocol is designed as an extended protocol compatible with ERC-721.  
+
+The “2 extensions” refer to the diversification of IP licensing and remixing.  
+
+- IP licensing methods are diverse. For example, delegating an NFT to someone else is one type of licensing, setting a price for the number of usage rights is another type of licensing, and even pricing based on auction, AMM, or other pricing mechanisms can develop different licensing methods. Therefore, the license module is designed allowing various custom licensing methods.  
+
+- IP remixing rules are also diverse. When remixing multiple existing NFTs, whether to support ERC-1155, whether to limit the range of NFT selection, and whether the NFT is consumed after remixing, there is no standard. So, the remix module is designed to support custom remixing rules.  
+
+The “1 constant” is that the traceability information of IP licensing is public and immutable. No matter how users license or remix IPs, the relationship between the original and new IPs stays the same. 
+
```

Comment:
The rationale section should go into detail about any technical choices made within the EIP itself. What you have so far is decent because it explains the overall design, but you should also include, for example, why you track relationships in a singleton instead of on the individual token contracts themselves.